packages/@cdktf/provider-generator/lib/get/generator/models/scope.ts

```diff
@@ -33,13 +34,39 @@ export class Scope {
   }
 
   public fullName(attributeName: string) {
+    return `${this.baseName}.${attributeName}`;
+  }
+
+  public get baseName() {
     // eslint-disable-next-line @typescript-eslint/no-this-alias
     let scope: Scope | undefined = this;
-    let name = attributeName;
+    let name = "";
     while (scope) {
-      name = `${scope.name}.${name}`;
+      if (name === "") {
+        name = scope.name;
+      } else {
+        name = `${scope.name}.${name}`;
+      }
       scope = scope.parent;
     }
     return name;
   }
+
+  public get attributePath() {
```

Comment:
This seems to be used in place of `fullName`; is there a particular reason why that wasn't used?